src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -57,5 +83,142 @@ public void SendAsync_SimpleGet_Success()
                     "Expected cancellation requested on token attached to exception.");
             }
         }
+
+        [Fact]
+        public void GetAsyncAutomaticDecompression()
+        {
+            var handler = new HttpClientHandler();
+            if (!handler.SupportsAutomaticDecompression)
```

Comment:
For CoreFx the handler should ALWAYS support automatic decompression even if it has to do it itself rather than rely on the underlying stack to do it.

For example, for Windows native WinHTTP supports it but only on Windows 8 and beyond.  Since CoreFx supports Windows 7, we have "manual" code to do the automatic decompression in the WinHttpHandler.

So, the design principle for CoreFx HttpClientHandler is that it should always return TRUE for supporting this feature. For implementation, if libcurl supports it, then it is easy.  Otherwise additional code needs to be added to the Curlhandler to support it.